datahub-web-react/src/app/ingest/secret/SecretBuilderModal.tsx

```diff
@@ -73,38 +63,35 @@ export const SecretBuilderModal = ({ initialState, visible, onSubmit, onCancel }
                 initialValues={initialState}
                 layout="vertical"
                 onFieldsChange={() =>
-                    setCreateButtonEnabled(form.getFieldsError().some((field) => field.errors.length > 0))
+                    setCreateButtonEnabled(!form.getFieldsError().some((field) => field.errors.length > 0))
                 }
             >
                 <Form.Item label={<Typography.Text strong>Name</Typography.Text>}>
                     <Typography.Paragraph>
                         Give your secret a name. This is what you&apos;ll use to reference the secret from your recipes.
                     </Typography.Paragraph>
                     <Form.Item
-                        name="name"
+                        name={NAME_FIELD_NAME}
                         rules={[
                             {
                                 required: true,
                                 message: 'Enter a name.',
                             },
-                            { whitespace: true },
+                            { whitespace: false },
                             { min: 1, max: 50 },
+                            { pattern: /^[^\s\t${}\\,'"]+$/, message: 'This secret name is not allowed.' },
```

Comment:
just some special characters regex?

Reply:
Yes! We don't want to screw up the injection into the recipe process... We've seen this as an issue before.